wiki/modding/intro.md

```diff
@@ -1,94 +1,82 @@
+| `Plugin.cs` | The main file that is loaded for your mod. |
+| `AssemblyInfo.cs` | File information about your mod. This is mostly managed by Modding Tools. |
+| `PluginConfig.cs` | A template for enabling config for your mod. This is commented out by default. |
+| `BSPlugin1Controller.cs` | A generic MonoBehaviour for your mod. |
+
+### Edit your mod's Manifest
+Fill out the `manifest.json` file with your information.  
+The `name` and `id` keys are used to identify your mod. The ID should match the ID used when uploading your mod to BeatMods.  
+
+::: warning
+Do **not** remove the dependency on BSIPA. As of BSIPA v4.1 this is required for your mod to load.
 :::
 
-## Fixing References
-To do this, right click on `References` in the Solution Explorer, and select `Add Reference...`
-
-![Add Reference](~@images/modding/add-a-ref.png "Add Reference")
-
-This will open the Reference Manager window, and you can browse to find the DLL files that are missing.
-Most of these files will be located within `\<Beat Saber directory>\Beat Saber_Data\Managed`
-
-(For IPA, use `IllusionInjector` and `IllusionPlugin`. For BSIPA, `IPA.Loader` is all that you need)
-
-![Finding References](~@images/modding/dnspy-assembly.png "Finding References")
-
-![References Windows](~@images/modding/plugin-addreferences.png "References Windows")
+## Compiling
+Build your plugin with `Build -> Build Solution` or <kbd>CTRL + SHIFT + B</kbd>  
+Your compiled DLL should automatically be copied to the `Plugins` folder in your Beat Saber directory! This will be done for both debug and release builds.
 
-Some Libraries are located in the Plugins folder as well. Check there for any libraries you forgot to add.
+::: tip NOTE
+When you are ready to release your mod, select the `Release` option to make a Release build of your mod.
 
-The most popular libraries located in the Plugins folder include:
-* CustomUI
-* BSML
+Building in Release mode will generate a packaged `.zip` file ready to upload to BeatMods.
+:::
 
-**Can't find your Beat Saber directory?** See [install folder](/faq/install-folder.md).
+## Testing your mod in-game
+To test if your mod is loaded in-game, you will need to launch Beat Saber with the BSIPA Console enabled. Add `--verbose` as a launch argument and run the game.  
+For more information on launch arguments, see [here](/modding/#launch-args).
 
-## Compiling
-Once you have fixed all project references, on the top menu bar press `Build -> Build Solution` or <kbd>CTRL + SHIFT + B</kbd>
+When you launch the game, you should see BSIPA load your mod in the console window.
 
-Your compiled DLL should appear in the `\Bin\Debug` folder of your project, or copied over to your Plugins folder, depending on what template you use.
-You can then copy this DLL into the `Plugins` folder within your Beat Saber directory.
+![Testing console screenshot](~@images/modding/testing-console.png "Testing console screenshot")
```

Comment:
Finally, I'd like to propose a `Next steps` section which would pretty much just be a link dump that people can dive in for  further information.

This would contain things like:
- Mention the BSMG `#pc-mod-dev` channel for mod development help
- A link to dnSpy for Beat Saber decompilation
- A link to BSIPA Docs for the Config system
- A link to Harmony for intermediate developers interested in method patching
- Zenject for experienced developers who know of Dependency Injection, since Beat Saber *heavily* uses this library.
  - While you're at it, might as well mention SiraUtil too, since that allows developers to easily hook into Beat Saber's Dependency Injection system.

Reply:
This looks good.

Reply:
How does the next steps writeup look?

Reply:
> For experienced developers, you may be interested learning about Zenject for Dependency Injection which the game heavily uses.

I think this could be worded better, something like `For experienced developers, you may be interested in learning about Zenject, the Dependency Injection system used heavily by Beat Saber.` The second sentence is fine.

My only other nitpick is that you should probably add a link to [Harmony](https://github.com/pardeike/Harmony) for patching the games code

Reply:
Late night me skipped the harmony bullet point... It has been added

Reply:
Correct, Harmony will come bundled with BSIPA